File: src/org/jetbrains/agda/core/CDataDeclaration.java
package org.jetbrains.agda.core;

import java.util.ArrayList;
import java.util.List;

/**
 * @author Evgeny.Kurbatsky
 */
public class CDataDeclaration extends CDeclaration {
    private String myName;
    private CExpression myType;
    private List<CTypeSignature> myConstructors = new ArrayList<CTypeSignature>();

    public CDataDeclaration(String name, CExpression type) {
        myName = name;
        myType = type;
    }

    public String getName() {
        return myName;
    }

    public CExpression getType() {
        return myType;
    }

    public List<CTypeSignature> geConstructors() {
        return myConstructors;
    }

    @Override
    public String toString() {
        StringBuilder builder = new StringBuilder();
        builder.append("data ").append(myName).append(" : ").append(myType).append(" {\n");
        for (CTypeSignature cConstructor : myConstructors) {
            builder.append("\t").append(cConstructor).append("\n");
        }
        builder.append("}");
        return builder.toString();
    }
}

File: src/org/jetbrains/agda/core/CDataDeclaration.kt
package org.jetbrains.agda.core

import java.util.ArrayList

/**
 * @author Evgeny.Kurbatsky
 */
public class CDataDeclaration(val name : String, aType : CExpression) : CDeclaration(aType) {
    private var myConstructors : MutableList<CTypeSignature> = ArrayList<CTypeSignature>()

    public fun getConstructors() : MutableList<CTypeSignature> {
        return myConstructors
    }
    public fun toString() : String? {
        val builder = StringBuilder()
        builder.append("data ").append(name).append(" : ").append(aType).append(" {\n")
        for (cConstructor : CTypeSignature? in myConstructors)
        {
            builder.append("\t").append(cConstructor).append("\n")
        }
        builder.append("}")
        return builder.toString()
    }
}
